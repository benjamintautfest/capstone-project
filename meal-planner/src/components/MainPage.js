import styled from 'styled-components/macro'
import Button from './Button'
import MealSelectMenu from './MealSelectMenu'
import RecipePage from './RecipePage'
import useRecipe from '../hooks/useRecipe'

export default function MainPage() {
    const {
        weekdays,
        selectedWeekday,
        showRecipe,
        selectedMeal,
        handleClick,
        handleCloseRecipe,
        handleShowRecipe,
        selectMeal,
    } = useRecipe()

    return (
        <SectionStyled>
            {showRecipe ? (
                <RecipePage
                    closeRecipe={handleCloseRecipe}
                    title={selectedMeal.title}
                    image={selectedMeal.image}
                    ingredients={selectedMeal.ingredients}
                    instructions={selectedMeal.instructions}
                />
            ) : (
                ''
            )}
            {weekdays &&
                weekdays.map(({ weekday, id, meal }) => (
                    <div key={id}>
                        <Button
                            day={weekday}
                            id={weekday}
                            meal={meal}
                            onClick={handleClick}
                        />
                        {selectedWeekday === weekday && (
                            <MealSelectMenu
                                handleMealClick={selectMeal}
                                weekdayId={id}
                                handleRecipeClick={handleShowRecipe}
                            />
                        )}
                    </div>
                ))}
        </SectionStyled>
    )
}

const SectionStyled = styled.section`
    overflow-y: auto;
    padding-top: 10px;
    scrollbar-width: none;
    &::-webkit-scrollbar {
        display: none;
    }
`
