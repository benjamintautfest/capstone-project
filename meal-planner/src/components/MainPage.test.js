import MainPage from './MainPage'
import { render } from '@testing-library/react'
import user from '@testing-library/user-event'

describe('MainPage', () => {
    it('renders correctly', () => {
        const { container } = render(
            <MainPage
                weekdays={[
                    { weekday: 'Montag', meal: 'Gemüselasagne', id: '01' },
                ]}
            />
        )
        expect(container.firstChild).toMatchSnapshot()
    })

    it('shows submenu item after button click', () => {
        const { getByText, queryByText } = render(<MainPage />)
        const button = getByText('Montag')
        expect(queryByText('Gemüselasagne')).not.toBeInTheDocument()

        user.click(button)
        const listitem = getByText('Gemüselasagne')

        expect(listitem).toBeInTheDocument()
    })
})
