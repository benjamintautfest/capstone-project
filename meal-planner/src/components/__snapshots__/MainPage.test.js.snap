// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders correctly 1`] = `
<section
  class="MainPage__SectionStyled-jplvfi-0 gEeWgy"
>
  
  <div>
    <button
      class="Button__ButtonStyled-sc-102td2i-0 dfgSOm"
      data-testid="Montag"
      id="Montag"
    >
      <span
        class="day"
      >
        Montag
      </span>
      <span
        class="meal"
      >
        Gem√ºselasagne
      </span>
    </button>
  </div>
</section>
`;
